File: solutions/1.two-sum.py
#
# @lc app=leetcode id=1 lang=python
#
# [1] Two Sum
#

# @lc code=start
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        num_map = {}
        for i in range(len(nums)):
            if target - nums[i] in num_map:
                return [num_map[target - nums[i]], i]
            num_map[nums[i]] = i
        return []
        
# @lc code=end

File: solutions_java/hashmaps/1.two-sum.java
package solutions_java.hashmaps;
/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int i=0; i<nums.length; i++) {
          if (!map.containsKey(nums[i])) {
            map.put(target-nums[i], i);
          } else {
            return new int[]{map.get(nums[i]),i};
          }
        }
        return new int[]{};
    }
}
// @lc code=end

